src/bci_build/package/versions.py

```diff
@@ -0,0 +1,151 @@
+"""This module contains a simple mechanism to store the package versions for
+various code streams in a json file.
+
+We need to know the version of certain packages in the distribution to be able
+to hardcode it in READMEs as we cannot let the build service create READMEs for
+us (this is a hard to work around limitation of OBS and the
+replace_using_package_version service).
+
+Therefore we keep a single json file in the code with all packages and their
+version per code stream in the git repository of the following form:
+
+.. code-block:: json
+
+   {
+       "nginx": {
+           "Tumbleweed": "1.25.5",
+           "6": "1.21.5",
+           "5": "1.21.5"
+       }
+   }
+
+
+The package versions are available via the function :py:func:`get_pkg_version`.
+
+The json file is updated via the function :py:func:`run_version_update`. It can
+also be called directly via :command:`poetry run update-versions` and is also
+run in the CI via :file:`.github/workflows/update-versions.yml`. Note that
+:py:func:`run_version_update` reads your OBS credentials from the environment
+variables ``OSC_USER`` and ``OSC_PASSWORD``. The function fails if one of them
+is not present.
+
+To add a new package to the versions json, add it's package name and all code
```

Comment:
Could the script create this automatically? The code knows all containers and all OS versions.

Reply:
can't you just loop over `ALL_CONTAINER_IMAGE_NAMES` and print the name/version with an empty version?

Reply:
That creates a circular import

Reply:
ok, in that case, it makes sense to manually add just what you need.